calculator-project/calculator.js

```diff
@@ -1,133 +1,57 @@
-    });
-
-    $("#btn8").click(function() {
-       resultArray.push(8);
-       $("#result").html(resultArray);
-    });
-
-    $("#btn9").click(function() {
-       resultArray.push(9);
-       $("#result").html(resultArray);
-    });
-
-    $("#btnPlus").click(function() {
-       resultArray.push("+");
-       $("#result").html(resultArray);
-    });
-
-    $("#btnMinus").click(function() {
-       resultArray.push("-");
-       $("#result").html(resultArray);
-    });
-
-    $("#btnDivide").click(function() {
-       resultArray.push("/");
-       $("#result").html(resultArray);
-    });
-
-    $("#btnMultiply").click(function() {
-       resultArray.push("*");
-       $("#result").html(resultArray);
-    });
-
-    $("#btnPercent").click(function() {
-       resultArray.push("%");
-       $("#result").html(resultArray);
-    });
-
-    $("#btnDecimal").click(function() {
-       resultArray.push(".");
-       $("#result").html(resultArray);
-    });
-
-    function calculate() {
-
-      for( let i = 0; i < resultArray.length; i++) {
-
-        if(isNaN(parseInt(resultArray[i])) && (resultArray[i] !== ".") && (i !== resultArray.length -1 )) {
-            var operator = resultArray[i];
-            num1 = resultArray.splice(0, i).join("");
-            num2 = resultArray.splice(1, resultArray.length - 2).join("");
-
-            if( operator === "+") {
-              calculation = parseFloat(num1) + parseFloat(num2);
-            }
-            else if( operator === "-") {
-              calculation = parseFloat(num1) - parseFloat(num2);
-            }
-            else if( operator === "/") {
-              calculation = (parseFloat(num1) / parseFloat(num2)).toFixed(1);
-            }
-            else if( operator === "*") {
-              calculation = (parseFloat(num1) * parseFloat(num2)).toFixed(1);
-            }
-            else if( operator === "%") {
-              calculation = (parseFloat(num1) / 100);
-            }
-
-         }
-      }
+  // creating array that stores the numbers from button clicks
+  var resultArray = [];
+  //stores numbers from resultArray that are on the left of an oppertor
+  var num1;
+  //stores numbers from resultArray that are on the right of an oppertor
+  var num2;
+  //stores the result of the calculation
+  var calculation;
+
+  $(".btn").click(function() {
+    // What can you do here to make "push" a one stop shop?
+    $("#result").html(resultArray);
+ });
+
+  $("#btnclear").click(function() {
+     resultArray = [];
+     $("#result").empty();
+     console.log('button clear clicked');
+  });
+
+  function calculate() {
+
+    for( let i = 0; i < resultArray.length; i++) {
+
+      if(isNaN(parseInt(resultArray[i])) && (resultArray[i] !== ".") && (i !== resultArray.length -1 )) {
+          var operator = resultArray[i];
+          num1 = resultArray.splice(0, i).join("");
+          num2 = resultArray.splice(1, resultArray.length - 2).join("");
+
+          if( operator === "+") {
```

Comment:
Look into `switch` conditional statement as a replacement for this if statement
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch